examples/fdc3-chart-and-grid/js-datagrid/src/app/main-view/services/mock-data.service.ts

```diff
@@ -12,52 +13,67 @@ export class MockDataService{
   public subject: Subject<any> = new Subject<any>();
   public marketData: any;
   private market: Market;
-  private currentChannel: Channel | null;
-  private connected: Boolean = false;
-  private connecting: Promise<void>;
+  private intentResolution: IntentResolution;
+  private readonly viewChartIntent: string = "ViewChart"; 
 
   constructor(){
     this.market = new Market();
-    this.connecting = new Promise(async(resolve, reject) => {
-      try{
-        resolve(await this.checkFdc3Connection());
-      } catch(err) {
-        reject(err);
-      }
-    });
 
     interval(1000).subscribe(() => {
       this.marketData = this.market.generateNewMarketNumbers();
       this.subject.next(this.marketData);
     });
   }
 
-  private async checkFdc3Connection(): Promise<void> {
-    if(!this.connected) {
-      this.currentChannel = await window.fdc3.getCurrentChannel();
-      if (!this.currentChannel) {
-        await window.fdc3.joinUserChannel("default");
-      }
-      this.connected = true;
-    }
-  }
-
-  public async publishSymbolData(symbol: any|undefined): Promise<void> {
+  public async publishSymbolData(symbol: Symbol | undefined): Promise<void> {
     if(symbol){
-      await this.connecting;
+      let marketSymbol;
       //TODO: involve DataService to generate the data for the chart
-      let marketSymbol = this.market.createMarketSymbol(symbol);
-
+      if (symbol) {
+        marketSymbol = this.market.createMarketSymbol(symbol);
+      }
+      
       const context: Context = {
-        type: 'fdc3.instrument',
+        type: ContextTypes.Chart,
         id: {
           ticker: marketSymbol?.symbol,
           buyData: marketSymbol?.buy,
```

Comment:
I know this is not new code, but this buyData and sellData is not part of the standard and in this sense, it's a misuse.
As I see it, the intent we raise should only contain that it's a ViewChart intent, and the instrument info. The rest of the data should be transferred via other means (preferably via a standard contexts for the sake of this example, but I'm not quite sure which one would fit this purpose). This would also add the possibility to "spice up" the example with "live data". We might want to discuss how to follow up on this offline :)